nrf-hal-common/src/lpcomp.rs

```diff
@@ -0,0 +1,295 @@
+//! HAL interface for the LPCOMP peripheral.
+//!
+//! In System ON, the LPCOMP can generate separate events on rising and falling edges of a signal,
+//! or sample the current state of the pin as being above or below the selected reference.
+//! The block can be configured to use any of the analog inputs on the device.
+//! Additionally, the low power comparator can be used as an analog wakeup source from System OFF.
+//! The comparator threshold can be programmed to a range of fractions of the supply voltage
+//! or to use an external analog reference input pin.
+
+use {
+    crate::gpio::{p0::*, Floating, Input},
+    crate::pac::{
+        generic::Reg,
+        lpcomp::{extrefsel::EXTREFSEL_A, psel::PSEL_A, _EVENTS_CROSS, _EVENTS_DOWN, _EVENTS_UP},
+        LPCOMP,
+    },
+};
+
+/// A safe wrapper around the `LPCOMP` peripheral.
+pub struct LpComp {
+    lpcomp: LPCOMP,
+}
+
+impl LpComp {
+    /// Takes ownership of the `LPCOMP` peripheral, returning a safe wrapper.
+    pub fn new<P: LpCompInputPin>(lpcomp: LPCOMP, input_pin: &P) -> Self {
+        lpcomp.psel.write(|w| w.psel().variant(input_pin.ain()));
+        lpcomp
+            .refsel
+            .write(|w| w.refsel().bits(VRef::_4_8Vdd.into()));
```

Comment:
Why select a reference here? Is this just resetting the value of refsel? If so, `lpcomp.refsel.reset();` should work too.

Reply:
I just chose an arbitrary reasonable in-the-middle default value, to have an "expected" default behaviour... Could be omitted, i just thought it was a good start value :)

Reply:
Okay, that seems fine. The function docs should mention it though.